spec/unit/http/session_spec.rb

```diff
@@ -83,4 +83,60 @@ def check_connection?(session, service, ssl_context: nil)
       }.to raise_error(ArgumentError, "Unknown service westbound")
     end
   end
+
+  context 'when resolving using multiple resolvers' do
+    let(:session) { client.create_session }
+
+    it "resolves once per session"
+
+    it "prefers SRV records" do
+      Puppet[:use_srv_records] = true
+      Puppet[:srv_domain] = "srv.example.com"
+
+      expect(session.resolvers.map(&:class)).to eq([Puppet::HTTP::Resolver::SRV, Puppet::HTTP::Resolver::Settings])
+      expect(session.resolvers[0]).to receive(:resolve).and_return(nil).ordered
+      expect(session.resolvers[1]).to receive(:resolve).and_call_original.ordered
+
+      session.route_to(:ca)
+    end
+
+    it "next prefers server_list" do
+      Puppet[:use_srv_records] = true
+      Puppet[:srv_domain] = "srv.example.com"
+      Puppet[:server_list] = 'foo.example.com,bar.example.com'
+
+      expect(session.resolvers.map(&:class)).to eq([Puppet::HTTP::Resolver::SRV, Puppet::HTTP::Resolver::ServerList, Puppet::HTTP::Resolver::Settings])
+      expect(session.resolvers[0]).to receive(:resolve).and_return(nil).ordered
+      expect(session.resolvers[1]).to receive(:resolve).and_return(nil).ordered
+      expect(session.resolvers[2]).to receive(:resolve).and_call_original.ordered
```

Comment:
I think this is off by one too. We want resolver 1 to be called and return a non-stubbed service object. Resolver 2 should never be called so that expect can be dropped.